src/Examine/LuceneEngine/Providers/LuceneIndexer.cs

```diff
@@ -1818,7 +1829,17 @@ private void InitializeDirectory()
                 VerifyFolder(LuceneIndexFolder);
                 _directory = DirectoryTracker.Current.GetDirectory(LuceneIndexFolder);
             }
+
+         
             _directory = DirectoryTracker.Current.GetDirectory(LuceneIndexFolder, InvokeDirectoryFactory);
+            if (_directory is ExamineDirectory)
+            {
+                DocumentWriting += (sender, args) =>
```

Comment:
I don't understand this, so this cancels all document writing events? Also we don't need to subscribe to our own events, this class is in charge of raising this event so it shouldn't subscribe to itself. Any changes needed should just be where the event is raised or where OnDocumentWriting is called

Reply:
I just notice I done that rule incorrectly, it should be added if readonly, and yes we cancel any writing to index, only master index should manage index, not readonly ones.